File: go/hanoi_tower.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"
)

var moveCount int

func hanoiTower(n int, from, to, aux rune) {
	if n == 1 {
		moveCount++
		return
	}
	
	hanoiTower(n-1, from, aux, to)
	moveCount++
	hanoiTower(n-1, aux, to, from)
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: hanoi_tower <n>")
		return
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil {
		fmt.Println("Invalid number:", os.Args[1])
		return
	}

	moveCount = 0

	start := time.Now()
	hanoiTower(n, 'A', 'C', 'B')
	elapsed := time.Since(start)

	timeMs := elapsed.Milliseconds()

	fmt.Printf("Go: hanoi_tower(%d) = %d\n", n, moveCount)
	fmt.Printf("Time: %dms\n", timeMs)
}

File: c/hanoi_tower.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int move_count = 0;

void hanoi_tower(int n, char from, char to, char aux) {
    if (n == 1) {
        move_count++;
        return;
    }
    
    hanoi_tower(n - 1, from, aux, to);
    move_count++;
    hanoi_tower(n - 1, aux, to, from);
}

int main(int argc, char* argv[]) {
    if (argc < 2) {
        printf("Usage: hanoi_tower <n>\n");
        return 1;
    }
    
    int n = atoi(argv[1]);
    move_count = 0;
    
    clock_t start = clock();
    hanoi_tower(n, 'A', 'C', 'B');
    clock_t end = clock();
    
    double time_ms = ((double)(end - start) / CLOCKS_PER_SEC) * 1000;
    
    printf("C: hanoi_tower(%d) = %d\n", n, move_count);
    printf("Time: %.0fms\n", time_ms);
    
    return 0;
}